src/server/game/Accounts/MuteMgr.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3
+ * Copyright (C) 2021+ WarheadCore <https://github.com/WarheadCore>
+ */
+
+#include "MuteMgr.h"
+#include "Chat.h"
+#include "DatabaseEnv.h"
+#include "Language.h"
+#include "ObjectMgr.h"
+#include "Player.h"
+#include "Timer.h"
+#include "World.h"
+
+MuteMgr* MuteMgr::instance()
+{
+    static MuteMgr instance;
+    return &instance;
+}
+
+void MuteMgr::MutePlayer(std::string const& targetName, uint32 notSpeakTime, std::string const& muteBy, std::string const& muteReason)
```

Comment:
Can't we use std::chrono instead of uint32 for notSpeakTime?

Also I propose changing the name to muteTime, feels better imo

Reply:
Hm, nice nice, okay